lib/src/browser/web_compiler.dart

```diff
@@ -0,0 +1,12 @@
+@JS()
+library web_compiler;
+
+import 'package:js/js.dart';
+
+@JS(r'$dartLoader')
+external dynamic get _dartLoader;
+
+// The compiler used to compile the Dart to JS, determined at runtime
+// via a global window property that is only present when DDC is used.
+// If false, DDC was used.
+bool get isCompiledWithDart2Js => _dartLoader == null;
```

Comment:
Could we add a test for this? Also, this is just semantics, but it might be more accurate to expose this as a `bool get isCompiledWithDdc => _dartLoader != null;` since we're using the presence of a DDC construct, not a dart2js one.

Reply:
Hmm. I don't know if there's really a way to test it without using it since the expectation would have to be set based on which compiler is being used... but to know which compiler is being used requires using the boolean being tested. We could assume that the test command in `tool/test.sh` is what'll always be used, but since `pub run test` uses dart2js, I don't think it's very useful to test that the property is null. Plus, if anyone used DDC the test would fail. 

It's easily manually tested, but if it ever changes, consumer CI will fail if people skip tests using it (like we'll have to in our big closed-source consumer). I'll add a more cautionary comment about using it and unfortunately I think that's the best we can do, unless you have any other ideas?